### PR TITLE
ARROW-8432: [CI] Don't depend on a single apache mirror for dependencies

### PR DESCRIPTION
**Pull request by bkietz**

### CODE REVIEW

File: dev/tasks/dependencies/download-apache.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# TODO(bkietz) reuse this
+# python/manylinux1/scripts/build_thrift.sh
+# python/manylinux201x/scripts/build_thrift.sh
```
Comment (bkietz) [inline dev/tasks/dependencies/download-apache.sh:21]: @kszucs any thoughts on how to do this?

**Comment by kszucs** [comment]
@github-actions crossbow submit test-ubuntu-18.04-docs test-conda-python-3.7-hdfs-2.9.2

File: ci/docker/conda-python-hdfs.dockerfile [hunk 2]
```diff
@@ -33,7 +33,10 @@ ARG hdfs=2.9.2
 ENV HADOOP_HOME=/opt/hadoop-${hdfs} \
     HADOOP_OPTS=-Djava.library.path=/opt/hadoop-${hdfs}/lib/native \
     PATH=$PATH:/opt/hadoop-${hdfs}/bin:/opt/hadoop-${hdfs}/sbin
-RUN wget -q -O - "https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=hadoop/common/hadoop-${hdfs}/hadoop-${hdfs}.tar.gz" | tar -xzf - -C /opt
+COPY ci/scripts/util_download_apache.sh /arrow/ci/scripts/
```
Comment (kszucs) [inline ci/docker/conda-python-hdfs.dockerfile:36]: To remain consistent with the rest of the dockerfiles.

File: ci/scripts/util_download_apache.sh [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+tarball_path=$1
+
+APACHE_MIRRORS=(
+  "https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename="
```
Comment (kszucs) [inline ci/scripts/util_download_apache.sh:22]: Added the dyn mirror.

**Comment by kszucs** [comment]
@github-actions crossbow submit test-ubuntu-18.04-docs test-conda-python-3.7-hdfs-2.9.2

**Comment by kszucs** [comment]
I'm building the manylinux images on one of the DGX machines to check thrift downloading works properly.

**Comment by kszucs** [comment]
Forgot that docker doesn't follow symlinks outside of the context, which are the `manylinux*` directories for each manylinux variant.

File: python/manylinux1/scripts/build_thrift.sh [hunk 4]
```diff
@@ -17,10 +17,7 @@
 # under the License.
 
 export THRIFT_VERSION=0.12.0
-THRIFT_DOWNLOAD_PATH=thrift/${THRIFT_VERSION}/thrift-${THRIFT_VERSION}.tar.gz
-curl -sL \
-  "https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename=${THRIFT_DOWNLOAD_PATH}" \
-  -o thrift-${THRIFT_VERSION}.tar.gz
+wget https://archive.apache.org/dist/thrift/${THRIFT_VERSION}/thrift-${THRIFT_VERSION}.tar.gz
```
Comment (kszucs) [inline python/manylinux1/scripts/build_thrift.sh:20]: The other manylinux images are alredy using the dist url. We rarely re-build this image so I think it's fine to keep as is.

**Comment by kszucs** [comment]
@github-actions crossbow submit test-ubuntu-18.04-docs test-conda-python-3.7-hdfs-2.9.2

**Comment by kszucs** [comment]
@github-actions crossbow submit test-ubuntu-18.04-docs test-conda-python-3.7-hdfs-2.9.2

**Comment by kszucs** [comment]
@github-actions crossbow submit test-ubuntu-18.04-docs test-conda-python-3.7-hdfs-2.9.2

**Comment by wesm** [comment]
Couldn't we also solve this problem by hosting a tarball on GitHub?

**Comment by kszucs** [comment]
We can add that as a follow-up, don't want to block the release further.

**Comment by kszucs** [comment]
Created JIRA https://issues.apache.org/jira/browse/ARROW-8454

**Comment by bkietz** [comment]
+1

**Pull request closed without merging**